### PR TITLE
Index

### PR DESCRIPTION
**Pull request by fs446**
Preparation for summer semester 2022 requires some mods as this material was not updated for a while.

Hence,
- `index.ipynb` has a Anconda env section
- `requirements.txt` was updated, especially requiring latest `sfs` toolbox
- `Dockerfile` was deleted as mybinder.org does not use this and does not build the repo anymore

### CODE REVIEW

**Comment by mgeier** [comment]
> * `Dockerfile` was deleted as mybinder.org does not use this and does not build the repo anymore

I like having a `.binder` directory, which allows specifying all `apt-get` requirements as well as Python packages.
For example:

* https://github.com/mgeier/python-audio/tree/master/.binder
* https://github.com/spatialaudio/nbsphinx/tree/master/.binder
* https://github.com/AudioSceneDescriptionFormat/splines/tree/master/.binder

**Comment by mgeier** [comment]
> * `Dockerfile` was deleted as mybinder.org does not use this and does not build the repo anymore

I like having a `.binder` directory, which allows specifying all `apt-get` requirements as well as Python packages.
For example:

* https://github.com/mgeier/python-audio/tree/master/.binder
* https://github.com/spatialaudio/nbsphinx/tree/master/.binder
* https://github.com/AudioSceneDescriptionFormat/splines/tree/master/.binder

File: index.ipynb [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "\n",
     "Registered students can access course details via [StudIP](https://studip.uni-rostock.de/dispatch.php/course/overview?cid=1df474b51c0813ffdeb34be5e66ef960) and [LSF](https://lsf.uni-rostock.de/qisserver/rds?state=verpublish&status=init&vmfile=no&publishid=78325&moduleCall=webInfo&publishConfFile=webInfo&publishSubDir=veranstaltung).\n",
     "\n",
-    "The lectures/exercises are held each summer semester, starting in 2015.\n",
+    "The lectures/exercises are held each summer semester, starting in 2015. Note, that we are currently (summer term 2022) modifying the notebooks to work with recent Python...\n",
```
Comment (mgeier) [inline index.ipynb]: I think this additional sentence isn't helping anyone. And recent Python isn't really the problem, it's rather all the third-party packages, isn't it?

File: index.ipynb [hunk 2]
```diff
@@ -82,7 +82,21 @@
...
+    "\n",
+    "### Anconda Usage\n",
+    "\n",
+    "If we want to utilize the [Anaconda](https://www.anaconda.com/products/distribution) distribution platform for Python, we can create a dedicated environment `mycomac` (we can also use a more creative name) for this course.\n",
+    "\n",
+    "For summer term 2022, we use\n",
+    "\n",
+    "- `conda --version`: 4.12.0\n",
+    "- `conda-build --version`: 3.21.8\n",
+    "- `conda create -n mycomac python=3.9 pip numpy sympy scipy matplotlib notebook ipykernel jupyterlab ipympl pydocstyle pycodestyle autopep8 flake8`\n",
```
Comment (mgeier) [inline index.ipynb]: I think it would be much better to provide a `environment.yml` instead.

File: index.ipynb [hunk 3]
```diff
@@ -82,7 +82,21 @@
...
+    "- or enforcing exact version numbers as `conda create -n mycomac python=3.9.12 pip=22.0.4 numpy=1.22.3 sympy=1.10.1 scipy=1.8.0 matplotlib=3.5.1 notebook=6.4.10 ipykernel=6.13.0 jupyterlab=3.3.4 ipympl=0.9.0 pydocstyle=6.1.1 pycodestyle=2.8.0 autopep8=1.6.0 flake8=4.0.1`\n",
```
Comment (mgeier) [inline index.ipynb]: I you want, you can use those versions in `environment.yml`, no need to confuse the students with that.

File: index.ipynb [hunk 4]
```diff
@@ -82,7 +82,21 @@
...
+    "- `pip install soundfile`\n",
+    "- `pip install sounddevice`\n",
+    "- `pip install sfs`\n",
```
Comment (mgeier) [inline index.ipynb]: Those things can also go into `environment.yml`.

File: index.ipynb [hunk 5]
```diff
@@ -82,7 +82,21 @@
...
+    "- we could archived the package version information of this environment, e.g. as `conda env export --no-builds > detailed_packages_list_yml.txt`"
```
Comment (mgeier) [inline index.ipynb]: I'm not sure how this would help students.

File: index.ipynb [hunk 6]
```diff
@@ -82,7 +82,21 @@
...
-    "\n"
+    "\n",
+    "### Anconda Usage\n",
+    "\n",
+    "If we want to utilize the [Anaconda](https://www.anaconda.com/products/distribution) distribution platform for Python, we can create a dedicated environment `mycomac` (we can also use a more creative name) for this course.\n",
+    "\n",
+    "For summer term 2022, we use\n",
+    "\n",
+    "- `conda --version`: 4.12.0\n",
+    "- `conda-build --version`: 3.21.8\n",
```
Comment (mgeier) [inline index.ipynb]: What do you need `conda-build` for?

File: index.ipynb (hunk 1 above)
Comment (fs446) [reply index.ipynb]: True indeed, it is not Python itself but rather libs. Removed.

File: index.ipynb (hunk 2 above)
Comment (fs446) [reply index.ipynb]: Yeah, that's reasonable, I provided `environment.yml` in the root directory.
Also added `.binder` folder with mybinder config information (thanks for this hint above!)

File: index.ipynb (hunk 3 above)
Comment (fs446) [reply index.ipynb]: Yes, in the `environment.yml` file I decided to use the version numbers from the environment I currently use to maintain the notebooks in order to have the version numbers somehow tracked.
Though, the `.binder` folder configuration is without explicit version requirement (except sfs), to have it more generic.
Does that make sense?

File: index.ipynb (hunk 4 above)
Comment (fs446) [reply index.ipynb]: Ok.

File: index.ipynb (hunk 5 above)
Comment (fs446) [reply index.ipynb]: Removed as `yml` is indeed nice for this issue already.

File: index.ipynb (hunk 6 above)
Comment (fs446) [reply index.ipynb]: Just tracked this as well out of habit. Removed.
The whole Anaconda text passage is now adapted towards handling with `environment.yml`

**Pull request merged**